midealocal/devices/da/__init__.py

```diff
@@ -137,30 +137,35 @@ def process_message(self, msg: bytes) -> dict[str, Any]:
         ]
         for status in self._attributes:
             if hasattr(message, str(status)):
+                value = getattr(message, str(status))
                 if status == DeviceAttributes.progress:
-                    self._attributes[status] = progress[getattr(message, str(status))]
+                    self._attributes[status] = (
+                        None if value >= len(progress) else progress[value]
+                    )
                 elif status == DeviceAttributes.program:
-                    self._attributes[status] = program[getattr(message, str(status))]
+                    self._attributes[status] = (
+                        None if value >= len(program) else program[value]
+                    )
                 elif status == DeviceAttributes.rinse_level:
-                    temp_rinse_level = getattr(message, str(status))
-                    if temp_rinse_level == MIN_TEMP:
-                        self._attributes[status] = "-"
-                    else:
-                        self._attributes[status] = temp_rinse_level
+                    self._attributes[status] = "-" if value == MIN_TEMP else value
```

Comment:
Didn't we voted for `None` ?

Reply:
I did just return to "-" for rinse_level, to keep the current behaviour.